cmd/nimbess-cni.go

```diff
@@ -164,7 +174,54 @@ func cmdAdd(args *skel.CmdArgs) error {
 		CNIVersion: conf.CNIVersion,
 	}
 
-	// TODO: handle IPAM type and data
+	// Run the IPAM plugin and get back the config to apply
+	ipamRequest, err := ipam.ExecAdd(cniRequest.IpamType, args.StdinData)
+	if err != nil {
+		return err
+	}
+
+	// Invoke ipam del if err to avoid ip leak
+	defer func() {
+		if err != nil {
+			ipam.ExecDel(cniRequest.IpamType, args.StdinData)
+		}
+	}()
+
+	// Convert whatever the IPAM result was into the current Result type
+	ipamResult, err := cnitypes.NewResultFromResult(ipamRequest)
+	if err != nil {
+		return err
+	}
+
+	if len(ipamResult.IPs) == 0 {
+		return errors.New("IPAM plugin returned missing IP config")
+	}
+
+	cniResult.IPs = ipamResult.IPs
+	cniResult.Routes = ipamResult.Routes
+
+	for _, ipc := range cniResult.IPs {
+		// All addresses apply to the container Nimbess interface
```

Comment:
maybe move this comment before the loop? Also maybe indicate why 0 indicates it "applies"